db/db_wal_test.cc

```diff
@@ -756,7 +756,8 @@ class RecoveryTestHelper {
   // Manuall corrupt the specified WAL
   static void CorruptWAL(DBWALTest* test, const Options& options,
                          const double off, const double len,
-                         const int wal_file_id, const bool trunc = false) {
+                         const int wal_file_id, MockEnv* mem_env,
+                         const bool trunc = false) {
```

Comment:
The change in db_wal_test is good, but I think if it doesn't run too long, we should make it run out of MockEnv. WAL uses different Env API sequential file that other part of DB don't. We need to get the coverage of the posix implementation for that.